src/formatDate.js

```diff
@@ -8,7 +8,46 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrayFromDate = date.split(fromFormat[fromFormat.length - 1]);
+  const arrForResulf = [];
+  const objectFromDate = {};
+
+  for (let i = 0; i < arrayFromDate.length; i++) {
+    objectFromDate[fromFormat[i]] = arrayFromDate[i];
+  }
+
+  const { MM: month, DD: day, ...lastElement } = objectFromDate;
+  const [year] = Object.values(lastElement);
+
+  for (const y of toFormat) {
+    if (y === 'YYYY' || y === 'YY') {
+      if (year.length === y.length) {
+        arrForResulf.push(year);
+      }
+
+      if (year.length > y.length) {
+        arrForResulf.push(year.slice(-2));
```

Comment:
The logic here assumes that a longer year string should be truncated to two digits. This might not be the intended behavior if the input date is already in a two-digit format. Consider validating the input date format before processing.